control/tests/nichols_test.py

```diff
@@ -26,3 +28,73 @@ def test_nichols(tsys, mplcleanup):
 def test_nichols_alias(tsys, mplcleanup):
     """Test the control.nichols alias and the grid=False parameter"""
     nichols(tsys, grid=False)
+
+
+class TestNicholsGrid:
```

Comment:
Could you please add the mplcleanup fixture here or for the whole module?

See e.g.
https://github.com/python-control/python-control/blob/46b3c53f752ccc7359083e3347da82b6020c80e0/control/tests/phaseplot_test.py#L21-L22

or

https://github.com/python-control/python-control/blob/46b3c53f752ccc7359083e3347da82b6020c80e0/control/tests/freqresp_test.py#L22

Reply:
This should also get rid of the necessity for `plt.clf()`

Reply:
Thanks, that is tidier.